app/models/user.rb

```diff
@@ -6,5 +6,7 @@ class User < ApplicationRecord
 
   enum role: { regular: 0, merchant: 1, admin: 2 }
 
+  has_one :merchant_employee
+  has_one :merchant, through: :merchant_employee
   has_secure_password
 end
```

Comment:
I'm curious, does this break anything else? I'm not sure that all users have merchants, but I'm also not sure how you would deal with this situation